assignments/closure.js

```diff
@@ -1,11 +1,34 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
+function sayName() {
+  let name = "Megan";
 
+  function greeting() {
+    let myName = "Caleb";
+    console.log(`Hey ${name}`);
+
+    function greetingFrom() {
+      console.log(`This is from ${myName}, in case you were wondering`);
+    }
+    greetingFrom();
+  }
+  greeting();
+}
+sayName();
 
 // ==== Challenge 2: Create a counter function ====
 const counter = () => {
```

Comment:
This is how I tweaked yours to have closure:

>const counter = () => {
  // Return a function that when invoked increments and returns a counter variable.
  let value = null;
  return function newCounter() {
    console.log((value += 1));
  };
};
const Acounter = counter();
console.log(Acounter());
console.log(Acounter());

The closure is created in Acounter, not counter(). Counter creates and destroys its scope whenever it is called. The Acounter needs to be set equal to the returned counter function outside of the counter function scope.